src/cable/connection.cr

```diff
@@ -52,36 +52,43 @@ module Cable
       @connection_rejected = true
     end
 
+    def channels : Array(Channel)
+      return Array(Channel).new unless Connection::CHANNELS.has_key?(connection_identifier)
+      Connection::CHANNELS.[connection_identifier].values
+    end
+
     def closed? : Bool
       socket.closed?
     end
 
     def close
-      return true unless Connection::CHANNELS.has_key?(connection_identifier)
+      if Connection::CHANNELS.has_key?(connection_identifier)
```

Comment:
We shouldn't return before closing the connection socket in the case of a connection without channel subscriptions.